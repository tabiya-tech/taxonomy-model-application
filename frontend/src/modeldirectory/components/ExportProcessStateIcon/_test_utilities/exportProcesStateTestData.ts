import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import ExportProcessStateAPISpecs from "api-specifications/importProcessState";
import { getMockId } from "src/_test_utilities/mockMongoId";

export const getAllExportProcessStatePermutations = (): ModelInfoTypes.ExportProcessState[] => {
  // generate all permutations of ExportProcessState status, errored, exportErrors, exportWarnings
  const allStatuses = Object.values(ExportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  const allBooleans = [true, false];

  const allPermutations: ModelInfoTypes.ExportProcessState[] = [];
  allStatuses.forEach((status) => {
    allBooleans.forEach((errored) => {
      allBooleans.forEach((exportErrors) => {
        allBooleans.forEach((exportWarnings) => {
          allPermutations.push({
            id: getMockId(allPermutations.length + 1),
            status,
            result: {
              errored,
              exportErrors: exportErrors,
              exportWarnings: exportWarnings,
            },
            downloadUrl: "https://example.com",
            timestamp: new Date(),
          });
        });
      });
    });
  });
  return allPermutations;
};
