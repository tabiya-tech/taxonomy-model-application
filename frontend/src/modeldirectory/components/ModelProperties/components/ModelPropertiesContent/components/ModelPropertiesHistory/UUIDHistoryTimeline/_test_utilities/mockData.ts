import { ModelInfoTypes } from "src//modelInfo/modelInfoTypes";

export function getOneFakeUUIDHistoryDetail(i: number): ModelInfoTypes.ModelHistory {
  return {
    UUID: "e26aca13-aa00-4346-9e83-edaf59314378" + i,
    id: "5e12f537f079fbdcb4e61bf" + i,
    localeShortCode: "ZA",
    name: "Model" + i,
    version: "V1.1." + i,
  };
}

export function getFakeUUIDHistoryDetailsArray(length: number): ModelInfoTypes.ModelHistory[] {
  return Array.from({ length }, (_, i) => getOneFakeUUIDHistoryDetail(i));
}

export function getFakeUUIDHistoryDetailsArrayWithUnresolvedUUIDS(length: number): ModelInfoTypes.ModelHistory[] {
  return Array.from({ length }, (_, i) => {
    const details = getOneFakeUUIDHistoryDetail(i);
    details.id = null;
    details.name = null;
    details.version = null;
    details.localeShortCode = null;
    return details;
  });
}
