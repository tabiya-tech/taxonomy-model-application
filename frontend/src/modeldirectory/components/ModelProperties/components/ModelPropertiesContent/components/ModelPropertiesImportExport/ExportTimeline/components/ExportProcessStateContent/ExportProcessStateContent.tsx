import React from "react";
import Box from "@mui/material/Box";
import { useTheme } from "@mui/material";
import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import ExportStatusPropertyField from "../ExportStatusPropertyField/ExportStatusPropertyField";
import DurationPropertyField from "src/theme/PropertyFieldLayout/DurationPropertyField/DurationPropertyField";

interface ExportProcessStateContentProps {
  exportProcessState: ModelInfoTypes.ExportProcessState;
}

const uniqueId = "333ab3d5-61a7-4169-8af9-01fe57ed5673";

export const DATA_TEST_ID = {
  EXPORT_PROCESS_STATE_CONTENT: `export-process-state-content-${uniqueId}`,
  EXPORT_STATUS_FIELD: `export-status-${uniqueId}`,
  EXPORT_DURATION_FIELD: `export-duration-${uniqueId}`,
};

export const FIELD_ID = {
  EXPORT_STATUS_FIELD: `export-status-${uniqueId}`,
  EXPORT_DURATION_FIELD: `export-duration-${uniqueId}`,
};

/**
 * ExportProcessStateContent is responsible for showing the export process status and the duration
 * @param props
 * @constructor
 */

const ExportProcessStateContent: React.FC<ExportProcessStateContentProps> = (
  props: Readonly<ExportProcessStateContentProps>
) => {
  const theme = useTheme();

  return (
    <Box
      display="flex"
      flexDirection="column"
      gap={theme.tabiyaSpacing.md}
      paddingY={theme.tabiyaSpacing.lg}
      data-testid={DATA_TEST_ID.EXPORT_PROCESS_STATE_CONTENT}
    >
      <ExportStatusPropertyField
        exportProcessState={props.exportProcessState}
        fieldId={FIELD_ID.EXPORT_STATUS_FIELD+"-"+props.exportProcessState.id}
        data-testid={DATA_TEST_ID.EXPORT_STATUS_FIELD}
      />
      <DurationPropertyField
        label="Duration"
        firstDate={props.exportProcessState.createdAt}
        secondDate={props.exportProcessState.updatedAt}
        fieldId={FIELD_ID.EXPORT_DURATION_FIELD+"-"+props.exportProcessState.id}
        data-testid={DATA_TEST_ID.EXPORT_DURATION_FIELD}
      />
    </Box>
  );
};

export default ExportProcessStateContent;
