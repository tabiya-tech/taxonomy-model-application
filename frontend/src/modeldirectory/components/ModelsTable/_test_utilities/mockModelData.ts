import ModelInfoAPISpecs from "api-specifications/modelInfo";
import LocaleAPISpecs from "api-specifications/locale";
import ImportProcessStateAPISpecs from "api-specifications/importProcessState";
import ExportProcessStateAPISpecs from "api-specifications/exportProcessState";
import { ImportProcessStateEnums } from "api-specifications/importProcessState/enums";

import { v4 as uuidv4 } from "uuid";

import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import { getMockId } from "src/_test_utilities/mockMongoId";
import { getRandomLorem, getRandomString, getTestString } from "src/_test_utilities/specialCharacters";
import { faker } from "@faker-js/faker";
import { CELL_MAX_LENGTH } from "src/modeldirectory/components/ModelsTable/ModelsTable";

export function getOneFakeModel(id?: number): ModelInfoTypes.ModelInfo {
  const model = getArrayOfFakeModels(1)[0];
  model.id = getMockId(id ?? 1);
  return model;
}

export function getArrayOfFakeModels(count: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < count; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      UUIDHistory: [uuidv4()],
      name: `${i + 1}/${count} - ${getRandomLorem(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH)}`.substring(
        0,
        ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH
      ),
      locale: {
        UUID: uuidv4(),
        name: faker.location.country().substring(0, ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: faker.location
          .countryCode("alpha-3")
          .substring(0, LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description:
        i % 2 === 0
          ? getRandomLorem(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH)
          : getRandomLorem(CELL_MAX_LENGTH / 2), // 50% chance of long description
      released: i % 2 === 0, // 50% chance of released
      releaseNotes: faker.lorem.text().substring(0, ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: faker.system.semver().substring(0, ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      exportProcessState: [getOneFakeExportProcessState(i)],
      importProcessState: getOneFakeImportProcessState(i)
    };
    models.push(model);
  }
  return models;
}

export function getArrayOfFakeModelsMaxLength(count: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < count; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      UUIDHistory: [uuidv4()],
      name: `${i + 1}/${count} - ${getRandomLorem(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH)}`.substring(
        0,
        ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH
      ),
      locale: {
        UUID: uuidv4(),
        name: getRandomLorem(LocaleAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: getRandomLorem(LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description: getRandomLorem(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH),
      released: i % 2 === 0, // 50% chance of released
      releaseNotes: getRandomLorem(ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: getRandomLorem(ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      exportProcessState: [getOneFakeExportProcessState(i)],
      importProcessState: getOneFakeImportProcessState(i)
    };
    models.push(model);
  }
  return models;
}

export function getOneRandomModelMaxLength(): ModelInfoTypes.ModelInfo {
  return getArrayOfRandomModelsMaxLength(1)[0];
}

export function getArrayOfRandomModelsMaxLength(number: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < number; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      UUIDHistory: [uuidv4()],
      name: getRandomString(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
      locale: {
        UUID: uuidv4(),
        name: getRandomString(LocaleAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: getTestString(LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description: getTestString(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH),
      released: i % 2 === 0,
      releaseNotes: getTestString(ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: getTestString(ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      exportProcessState: [getOneFakeExportProcessState(i)],
      importProcessState: getOneFakeImportProcessState(i)
    };
    models.push(model);
  }
  return models;
}

export const fakeModel: ModelInfoTypes.ModelInfo = {
  UUID: "8d914eab-6f7d-4183-accc-a09f99887b39",
  UUIDHistory: ["8d914eab-6f7d-4183-accc-a09f99887b39"],
  description: "aw j zt   agrkasl dy ogtimpsauwumu l utrovthao syertm beawpxluhyudgzbbm",
  id: "000000000000000000000001",
  importProcessState: {
    id: "000000000000000000000001",
    result: {
      errored: true,
      parsingErrors: true,
      parsingWarnings: true,
    },
    status: ImportProcessStateEnums.Status.PENDING,
    createdAt: new Date("2023-10-18T17:35:10.571Z"),
    updatedAt: new Date("2023-10-18T17:35:12.571Z"),
  },
  exportProcessState: [
    {
      downloadUrl: "https://www.example.com/",
      id: "000000000000000000000001",
      result: {
        errored: true,
        exportErrors: true,
        exportWarnings: true,
      },
      status: ExportProcessStateAPISpecs.Enums.Status.PENDING,
      timestamp: new Date("2023-10-18T17:35:10.571Z"),
      createdAt: new Date("2023-10-18T17:35:10.571Z"),
      updatedAt: new Date("2023-10-18T17:35:12.571Z"),
    },
  ],
  locale: {
    UUID: "7c4ea00b-fbea-4a31-9c04-009c813f5f34",
    name: "Palestine",
    shortCode: "ETH",
  },
  name: "1/1 - fysmros tcgcnjbbrrev",
  path: "https://unwelcome-editorial.net/",
  releaseNotes: "Perferendis modi impedit necessitatibus a",
  released: true,
  tabiyaPath: "https://silver-vulture.biz",
  createdAt: new Date("2023-10-18T17:35:10.571Z"),
  updatedAt: new Date("2022-10-18T17:35:10.571Z"),
  version: "0.2.2",
};

export function getRandomImportStatus(id: number) {
  const allStatuses = Object.values(ImportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  return allStatuses[id % allStatuses.length];
}

export function getRandomExportStatus(id: number) {
  const allStatuses = Object.values(ExportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  return allStatuses[id % allStatuses.length];
}

export function getOneFakeExportProcessState(i: number): ModelInfoTypes.ExportProcessState {
  return {
    id: getMockId(10000 + i),
    status: getRandomExportStatus(i),
    result: {
      errored: i % 2 === 0,
      exportErrors: i % 2 === 0,
      exportWarnings: i % 2 === 0,
    },
    downloadUrl: faker.internet.url(),
    timestamp: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
    createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
    updatedAt: new Date(),
  };
}

export function getOneFakeImportProcessState(i: number): ModelInfoTypes.ImportProcessState {
  return {
    id: getMockId(10000 + i),
    status: getRandomImportStatus(i),
    result: {
      errored: i % 2 === 0,
      parsingErrors: i % 2 === 0,
      parsingWarnings: i % 2 === 0,
    },
    createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
    updatedAt: new Date(),
  };
}
