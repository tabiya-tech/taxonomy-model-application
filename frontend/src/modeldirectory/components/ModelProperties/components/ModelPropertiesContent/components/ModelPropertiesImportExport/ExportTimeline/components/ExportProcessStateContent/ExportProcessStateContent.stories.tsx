import { Meta, StoryObj } from "@storybook/react";
import { getOneFakeExportProcessState } from "src/modeldirectory/components/ModelsTable/_test_utilities/mockModelData";
import ExportProcessStateContent from "./ExportProcessStateContent";

const meta: Meta<typeof ExportProcessStateContent> = {
  title:
    "ModelDirectory/ModelProperties/ModelPropertiesContentPanels/ModelPropertiesImportExportComponents/ExportProcessStateContent",
  component: ExportProcessStateContent,
  tags: ["autodocs"],
  args: {
    exportProcessState: getOneFakeExportProcessState(1),
  },
};

type Story = StoryObj<typeof ExportProcessStateContent>;

export const Shown: Story = {};

export default meta;
