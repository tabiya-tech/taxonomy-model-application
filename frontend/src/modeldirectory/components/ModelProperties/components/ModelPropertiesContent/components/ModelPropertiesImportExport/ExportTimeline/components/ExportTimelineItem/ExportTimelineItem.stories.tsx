import { Meta, StoryObj } from "@storybook/react";
import { fakeModel } from "src/modeldirectory/components/ModelsTable/_test_utilities/mockModelData";
import ExportTimelineItem from "./ExportTimelineItem";
import { Timeline } from "@mui/lab";

const meta: Meta<typeof ExportTimelineItem> = {
  title:
    "ModelDirectory/ModelProperties/ModelPropertiesContentPanels/ModelPropertiesImportExportComponents/ExportTimelineItem",
  component: ExportTimelineItem,
  tags: ["autodocs"],
  args: {
    exportProcessState: fakeModel.exportProcessState[0],
  },
  decorators: [
    (Story) => (
      <Timeline>
        <Story />
      </Timeline>
    ),
  ],
};

type Story = StoryObj<typeof ExportTimelineItem>;

export const Shown: Story = {};

export default meta;
