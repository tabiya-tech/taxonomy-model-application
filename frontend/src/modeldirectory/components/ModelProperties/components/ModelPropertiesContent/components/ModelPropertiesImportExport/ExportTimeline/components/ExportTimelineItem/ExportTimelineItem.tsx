import React from "react";
import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import {
  TimelineConnector,
  TimelineContent,
  TimelineDot,
  TimelineItem,
  TimelineOppositeContent,
  TimelineSeparator,
} from "@mui/lab";
import { Typography, useTheme } from "@mui/material";
import ExportProcessStateContent from "src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ExportTimeline/components/ExportProcessStateContent/ExportProcessStateContent";
import { formatDate } from "src/theme/PropertyFieldLayout/FormattedDatePropertyField/userFriendlyDateFormat";
import DownloadModelButton from "src/modeldirectory/components/DownloadModelButton/DownloadModelButton";

interface ExportTimelineItemProps {
  exportProcessState: ModelInfoTypes.ExportProcessState;
}

const uniqueId = "e50b6e98-dd42-4e1b-b7c8-7a282f67b032";
export const DATA_TEST_ID = {
  EXPORT_TIMELINE_ITEM: `export-timeline-item-${uniqueId}`,
  EXPORT_TIMELINE_SEPARATOR: `export-timeline-separator-${uniqueId}`,
  EXPORT_TIMELINE_DOT: `export-timeline-dot-${uniqueId}`,
  EXPORT_TIMELINE_CONNECTOR: `export-timeline-connector-${uniqueId}`,
  EXPORT_TIMELINE_CONTENT: `export-timeline-content-${uniqueId}`,
  EXPORT_TIMELINE_OPPOSITE_CONTENT: `export-timeline-opposite-content-${uniqueId}`,
  EXPORT_TIMELINE_OPPOSITE_CONTENT_CREATED_AT: `export-timeline-opposite-content-created-at-${uniqueId}`,
};

/**
 * ExportTimelineItem is responsible for showing a timeline item with the export process state content and creation date
 */
const ExportTimelineItem: React.FC<ExportTimelineItemProps> = (props) => {
  const theme = useTheme();
  return (
    <TimelineItem data-testid={DATA_TEST_ID.EXPORT_TIMELINE_ITEM}>
      <TimelineOppositeContent
        data-testid={DATA_TEST_ID.EXPORT_TIMELINE_OPPOSITE_CONTENT}
        display={"flex"}
        flexDirection={"column"}
        alignItems={"flex-start"}
        gap={theme.tabiyaSpacing.md}
        sx={{ maxWidth: "35%", padding: 0, paddingRight: theme.tabiyaSpacing.md }}
      >
        <Typography
          variant={"body1"}
          data-testid={DATA_TEST_ID.EXPORT_TIMELINE_OPPOSITE_CONTENT_CREATED_AT}
          sx={{ textAlign: "start" }}
        >
          {formatDate(props.exportProcessState.createdAt)}
        </Typography>
        {props.exportProcessState.downloadUrl && (
          <DownloadModelButton downloadUrl={props.exportProcessState.downloadUrl} />
        )}
      </TimelineOppositeContent>
      <TimelineSeparator data-testid={DATA_TEST_ID.EXPORT_TIMELINE_SEPARATOR}>
        <TimelineDot
          data-testid={DATA_TEST_ID.EXPORT_TIMELINE_DOT}
          color={"primary"}
          sx={{ margin: (theme) => theme.tabiyaSpacing.xs }}
        />
        <TimelineConnector
          data-testid={DATA_TEST_ID.EXPORT_TIMELINE_CONNECTOR}
          sx={{ backgroundColor: theme.palette.primary.main }}
        />
      </TimelineSeparator>
      <TimelineContent data-testid={DATA_TEST_ID.EXPORT_TIMELINE_CONTENT}>
        <ExportProcessStateContent exportProcessState={props.exportProcessState} />
      </TimelineContent>
    </TimelineItem>
  );
};

export default ExportTimelineItem;
