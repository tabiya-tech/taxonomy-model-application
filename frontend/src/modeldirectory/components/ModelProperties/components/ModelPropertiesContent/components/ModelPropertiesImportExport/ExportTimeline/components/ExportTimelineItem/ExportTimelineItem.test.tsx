// mute chatty console
import "src/_test_utilities/consoleMock";

import { render, screen } from "src/_test_utilities/test-utils";
import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import { getOneFakeExportProcessState } from "src/modeldirectory/components/ModelsTable/_test_utilities/mockModelData";
import ExportTimelineItem, { DATA_TEST_ID } from "./ExportTimelineItem";
import { within } from "@testing-library/react";
import ExportProcessStateContent from "src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ExportTimeline/components/ExportProcessStateContent/ExportProcessStateContent";

// mock the ExportProcessStateContent component
jest.mock(
  "src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ExportTimeline/components/ExportProcessStateContent/ExportProcessStateContent",
  () => {
    const actual = jest.requireActual(
      "src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ExportTimeline/components/ExportProcessStateContent/ExportProcessStateContent"
    );
    const mockExportProcessStateContent = jest
      .fn()
      .mockImplementation(() => <div data-testid="export-process-state-content">ExportProcessStateContent</div>);
    return {
      __esModule: true,
      ...actual,
      default: mockExportProcessStateContent,
    };
  }
);

describe("ExportTimelineItem", () => {
  test("renders correctly with the provided exportProcessState", () => {
    // GIVEN an exportProcessState
    const givenExportProcessState: ModelInfoTypes.ExportProcessState = getOneFakeExportProcessState(1);

    const givenUserFriendlyFormattedDate = "foo-date-user-friendly-formatted";
    const formatDateSpy = jest
      .spyOn(require("src/theme/PropertyFieldLayout/FormattedDatePropertyField/userFriendlyDateFormat"), "formatDate")
      .mockReturnValue(givenUserFriendlyFormattedDate);
    // WHEN the ExportTimelineItem component is rendered with the given exportProcessState
    render(<ExportTimelineItem exportProcessState={givenExportProcessState} />);

    // THEN expect no errors or warning to have occurred
    expect(console.error).not.toHaveBeenCalled();
    expect(console.warn).not.toHaveBeenCalled();
    // AND the component to be displayed
    const exportTimelineItemComponent = screen.getByTestId(DATA_TEST_ID.EXPORT_TIMELINE_ITEM);
    expect(exportTimelineItemComponent).toBeInTheDocument();
    // AND the timelineContent component to be displayed
    const timelineContentComponent = screen.getByTestId(DATA_TEST_ID.EXPORT_TIMELINE_CONTENT);
    expect(timelineContentComponent).toBeInTheDocument();
    // AND the exportProcessStateContent component to be displayed in the timelineContent
    const exportProcessStateContentComponent =
      within(timelineContentComponent).getByTestId("export-process-state-content");
    expect(exportProcessStateContentComponent).toBeInTheDocument();
    // AND the ExportProcessStateContent to have been called with the correct props
    expect(ExportProcessStateContent).toHaveBeenCalledWith({ exportProcessState: givenExportProcessState }, {});
    // AND the timelineSeparator component to be displayed
    const timelineSeparatorComponent = screen.getByTestId(DATA_TEST_ID.EXPORT_TIMELINE_SEPARATOR);
    expect(timelineSeparatorComponent).toBeInTheDocument();
    // AND the timelineDot component to be displayed
    const timelineDotComponent = screen.getByTestId(DATA_TEST_ID.EXPORT_TIMELINE_DOT);
    expect(timelineDotComponent).toBeInTheDocument();
    // AND the timelineConnector component to be displayed
    const timelineConnectorComponent = screen.getByTestId(DATA_TEST_ID.EXPORT_TIMELINE_CONNECTOR);
    expect(timelineConnectorComponent).toBeInTheDocument();
    // AND the timelineOppositeContent component to be displayed
    const timelineOppositeContentComponent = screen.getByTestId(DATA_TEST_ID.EXPORT_TIMELINE_OPPOSITE_CONTENT);
    expect(timelineOppositeContentComponent).toBeInTheDocument();
    // AND the export creation date to be displayed in the timelineOppositeContent
    const createdAtComponent = within(timelineOppositeContentComponent).getByTestId(
      DATA_TEST_ID.EXPORT_TIMELINE_OPPOSITE_CONTENT_CREATED_AT
    );
    expect(createdAtComponent).toBeInTheDocument();
    // AND the export creation date to be the given user friendly formatted date
    expect(formatDateSpy).toHaveBeenCalledWith(givenExportProcessState.timestamp);
    // AND the component to match the snapshot
    expect(exportTimelineItemComponent).toMatchSnapshot();
  });
});
