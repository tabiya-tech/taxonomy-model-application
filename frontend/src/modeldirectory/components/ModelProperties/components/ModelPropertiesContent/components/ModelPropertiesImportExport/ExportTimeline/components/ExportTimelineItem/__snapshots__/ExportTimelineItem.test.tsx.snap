// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportTimelineItem renders correctly when the exportProcessState doesnt have a downloadUrl 1`] = `
<li
  class="MuiTimelineItem-root MuiTimelineItem-positionRight css-ny8qih-MuiTimelineItem-root"
  data-testid="export-timeline-item-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
>
  <div
    class="MuiTypography-root MuiTypography-body1 MuiTimelineOppositeContent-root MuiTimelineOppositeContent-positionRight css-189t2ml-MuiTypography-root-MuiTimelineOppositeContent-root"
    data-testid="export-timeline-opposite-content-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ghakyl-MuiTypography-root"
      data-testid="export-timeline-opposite-content-created-at-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
    >
      foo-date-user-friendly-formatted
    </p>
  </div>
  <div
    class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
    data-testid="export-timeline-separator-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
  >
    <span
      class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledPrimary css-12fjwjc-MuiTimelineDot-root"
      data-testid="export-timeline-dot-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
    />
    <span
      class="MuiTimelineConnector-root css-itego1-MuiTimelineConnector-root"
      data-testid="export-timeline-connector-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
    />
  </div>
  <div
    class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-p3xido-MuiTypography-root-MuiTimelineContent-root"
    data-testid="export-timeline-content-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
  >
    <div
      data-testid="export-process-state-content"
    >
      ExportProcessStateContent
    </div>
  </div>
</li>
`;

exports[`ExportTimelineItem renders correctly with the provided exportProcessState when user does not have anonymous role 1`] = `
<li
  class="MuiTimelineItem-root MuiTimelineItem-positionRight css-ny8qih-MuiTimelineItem-root"
  data-testid="export-timeline-item-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
>
  <div
    class="MuiTypography-root MuiTypography-body1 MuiTimelineOppositeContent-root MuiTimelineOppositeContent-positionRight css-189t2ml-MuiTypography-root-MuiTimelineOppositeContent-root"
    data-testid="export-timeline-opposite-content-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ghakyl-MuiTypography-root"
      data-testid="export-timeline-opposite-content-created-at-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
    >
      foo-date-user-friendly-formatted
    </p>
    <a
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-aqwr9w-MuiButtonBase-root-MuiIconButton-root"
      data-testid="download-model-button-1623572c-456c-419e-9201-3c4971406fe1"
      download="https://www.example.com/"
      href="https://www.example.com/"
      tabindex="0"
      title="Download"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-dy5dri-MuiSvgIcon-root"
        data-testid="FileDownloadIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
        />
      </svg>
      <span
        class="MuiTypography-root MuiTypography-caption css-1w1i5dg-MuiTypography-root"
      >
        Download
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
  <div
    class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
    data-testid="export-timeline-separator-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
  >
    <span
      class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledPrimary css-12fjwjc-MuiTimelineDot-root"
      data-testid="export-timeline-dot-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
    />
    <span
      class="MuiTimelineConnector-root css-itego1-MuiTimelineConnector-root"
      data-testid="export-timeline-connector-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
    />
  </div>
  <div
    class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-p3xido-MuiTypography-root-MuiTimelineContent-root"
    data-testid="export-timeline-content-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
  >
    <div
      data-testid="export-process-state-content"
    >
      ExportProcessStateContent
    </div>
  </div>
</li>
`;

exports[`ExportTimelineItem renders correctly with the provided exportProcessState when user has anonymous role 1`] = `
<li
  class="MuiTimelineItem-root MuiTimelineItem-positionRight css-ny8qih-MuiTimelineItem-root"
  data-testid="export-timeline-item-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
>
  <div
    class="MuiTypography-root MuiTypography-body1 MuiTimelineOppositeContent-root MuiTimelineOppositeContent-positionRight css-189t2ml-MuiTypography-root-MuiTimelineOppositeContent-root"
    data-testid="export-timeline-opposite-content-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ghakyl-MuiTypography-root"
      data-testid="export-timeline-opposite-content-created-at-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
    >
      foo-date-user-friendly-formatted
    </p>
  </div>
  <div
    class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
    data-testid="export-timeline-separator-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
  >
    <span
      class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledPrimary css-12fjwjc-MuiTimelineDot-root"
      data-testid="export-timeline-dot-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
    />
    <span
      class="MuiTimelineConnector-root css-itego1-MuiTimelineConnector-root"
      data-testid="export-timeline-connector-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
    />
  </div>
  <div
    class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-p3xido-MuiTypography-root-MuiTimelineContent-root"
    data-testid="export-timeline-content-e50b6e98-dd42-4e1b-b7c8-7a282f67b032"
  >
    <div
      data-testid="export-process-state-content"
    >
      ExportProcessStateContent
    </div>
  </div>
</li>
`;
